src/bokeh/models/widgets/tables.py

```diff
@@ -858,6 +859,54 @@ def __init__(self, *args, **kwargs) -> None:
     The height of each row in pixels.
     """)
 
+    @staticmethod
+    def from_data(data, columns=None, formatters={}, **kwargs) -> DataTable:
+        """ Create a simple table from a pandas dataframe, dictionary or ColumnDataSource.
+
+        Args:
+            data (DataFrame or dict or ColumnDataSource) :
+                The data to create the table from. If the data is a dataframe
+                or dictionary, a ColumnDataSource will be created from it.
+
+            columns (list, optional) :
+                A list of column names to use from the input data.
+                If None, use all columns. (default: None)
+
+            formatters (dict, optional) :
+                A mapping of column names and corresponding Formatters to
+                apply to each column. (default: None)
+
+        Keyword arguments:
+            Any additional keyword arguments will be passed to DataTable.
+
+        Returns:
+            DataTable
+
+        Raises:
+            ValueError
+                If the provided data is not a ColumnDataSource
+                or a data source that a ColumnDataSource can be created from.
+
+        """
+
+        if isinstance(data, ColumnDataSource):
+            source = data
```

Comment:
so the suggestion is that this should always be 
```
source = data.clone()
```
And then add a test to make sure that `table.source` and the CDS that was passed in are *not* the same object

Reply:
The `.clone` method returns a RuntimeError on my computer. I have no idea why(per my previous reply and screenshot).

Reply:
You need at least bokeh 3.4.0-dev2 for it to work (PR #13443).

Reply:
OK. Thank you. I tried it in my dev environment and it worked.